### ISSUE TITLE
[RAC] Key navigation is broken for nested GridLists

### ISSUE THREAD

**Issue by piecyk** [adobe/react-spectrum#5179]
### Provide a general summary of the issue here

After focusing on nested element, pressing arrow down/up will navigate to element in parent list.

### 🤔 Expected Behavior?

Key navigation should be trapped within nested list, till espaced with tab.

### 😯 Current Behavior

Pressing arrow down/up will navigate element in parent list.

### 🖥️ Steps to Reproduce

https://codesandbox.io/s/throbbing-wind-jflfv7?file=/src/App.js

### Version

"react-aria-components": "1.0.0-beta.0"

### What browsers are you seeing the problem on?

Chrome

### What operating system are you using?

MacOS

**Comment by piecyk** [comment]
It boils down that `useGridListItem` handles key events with `onKeyDownCapture` https://github.com/piecyk/react-spectrum/blame/main/packages/%40react-aria/gridlist/src/useGridListItem.ts#L184

Is there specific reason for that, would expect onKeyDown here 🤔

**Comment by LFDanLu** [comment]
If I remember correctly, part of the reason for the `onKeyDownCapture` is so that the GridList always handles the keyboard events for grid navigation, regardless of what focusable element might have focus within the row (e.g. it would handle the ArrowUp/ArrowDown over say a menu button).

**Comment by reidbarber** [comment]
I think supporting nested grid roles will take more to properly implement, and falls under: https://github.com/adobe/react-spectrum/issues/2328

**Comment by piecyk** [comment]
> If I remember correctly, part of the reason for the onKeyDownCapture is so that the GridList always handles the keyboard events for grid navigation, regardless of what focusable element might have focus within the row (e.g. it would handle the ArrowUp/ArrowDown over say a menu button).

OK, that make sense, this should work the same for nested list.

**Comment by devongovett** [comment]
Is your use case sections in the list? If so, I'm not sure if that should be implemented as nested grids. We were working on support for sections in our table and list components a few months back, and I believe they were all within the same list rather than using nested lists. @LFDanLu probably knows more about that.

**Comment by piecyk** [comment]
My use case is that use can create list of comments ( it's reorderable ) and to each comment can include a list of some items ( this is also reorderable ). in theory can flatten it, just when reordering the comments need to keep track of it items, to move it together 🤔

**Comment by LFDanLu** [comment]
> We were working on support for sections in our table and list components a few months back, and I believe they were all within the same list rather than using nested lists

That is correct, from the collection's perspective it was just a single table/list. The gridlists/table also had multiple row groups in the DOM rather than nested tables/gridlists.

**Comment by devongovett** [comment]
Ah ok, the reordering makes that more complicated. Our sections were not interactive. In that case this sounds more like a tree. We are currently working on tree grid support as well, but it's not quite ready yet.

**Comment by LFDanLu** [comment]
@piecyk Now that drag and drop support is available in Tree: https://react-spectrum.adobe.com/react-aria/Tree.html#drag-and-drop-, does that work for you instead of a gridlist?

**Comment by James-Wilkinson-git** [comment]
We ran into similar issues building a component library wrapper around RAC's tree grid (Table with nested Row elements). A few things that were required to get correct keyboard navigation:

1. Always wrap children in `<Collection items={columns}>` — even for static JSX children where columns is undefined. This is how RAC's collection system registers nested items against the correct parent context. Rendering children directly without Collection means child items register against the wrong ancestor and navigation leaks up to the parent, exactly as described here.

2. For Table tree grids specifically, the treeColumn prop on `<Table>` is required — without it, RAC doesn't know which column governs the tree and keyboard expand/collapse (Arrow Right/Left) never fires.

3. For the dynamic/recursive pattern, child items should be rendered via a separate `<Collection items={item.children}>{renderItem}</Collection>` inside the row JSX, not as static nested element children. This is the pattern in the RAC docs under "Expandable rows" and is what causes hasChildItems to resolve correctly on parent rows.

It's possible the same Collection registration issue affects nested GridList — worth checking whether items is undefined vs not being passed at all, and whether the children are going through Collection in the nested list.
